src/librustc/hir/mod.rs

```diff
@@ -2074,6 +2072,13 @@ pub struct InlineAsm {
     pub dialect: AsmDialect,
 }
 
+#[derive(RustcEncodable, RustcDecodable, Debug, HashStable)]
+pub struct InlineAsm {
+    pub inner: InlineAsmInner,
```

Comment:
Is there any need to have `InlineAsmInner`? Could we just have `asm`, `asm_str_style`, `outputs`, and `dialect` fields directly in `InlineAsm`?

Reply:
I tried but wasn't able to do that; `Expr` is not `Clone` and `InlineAsmInner` needs to be `Clone`.